src/gui/log/loglistview.cpp

```diff
@@ -0,0 +1,140 @@
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "loglistview.h"
+
+#include <QApplication>
+#include <QClipboard>
+#include <QFontMetrics>
+#include <QKeyEvent>
+#include <QPainter>
+#include <QStyle>
+#include <QStyledItemDelegate>
+
+#include "logmodel.h"
+#include "uithememanager.h"
+
+namespace
+{
+    const QString SEPARATOR = QStringLiteral(" - ");
+
+    int horizontalAdvance(const QFontMetrics &fontMetrics, const QString &text)
+    {
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 11, 0))
+        return fontMetrics.horizontalAdvance(text);
+#else
+        return fontMetrics.width(text);
+#endif
+    }
+
+    QString logText(const QModelIndex &index)
+    {
+        return QString::fromLatin1("%1%2%3").arg(index.data(BaseLogModel::TimeRole).toString(), SEPARATOR
+                                                 , index.data(BaseLogModel::MessageRole).toString());
+    }
+
+    class LogItemDelegate : public QStyledItemDelegate
+    {
+    public:
+        using QStyledItemDelegate::QStyledItemDelegate;
+
+    private:
+        void paint(QPainter *painter, const QStyleOptionViewItem &option, const QModelIndex &index) const override
+        {
+            painter->save();
+            QStyledItemDelegate::paint(painter, option, index); // paints background, focus rect and selection rect
+
+            const QStyle *style = option.widget ? option.widget->style() : QApplication::style();;
+            const QRect textRect = style->subElementRect(QStyle::SE_ItemViewItemText, &option, option.widget)
+                                   .adjusted(1, 0, 0, 0); // shift 1 to avoid text being too close to focus rect
+
+            QFont font = option.font;
+            if (option.font.pointSize() > 0)
+                font.setPointSize(option.font.pointSize()); // somehow this needs to be set directly else painter will use default font
+            painter->setFont(font);
+
+            const QPen originalPen = painter->pen();
+            QPen coloredPen = originalPen;
+            coloredPen.setColor(Qt::darkGray);
+            painter->setPen(coloredPen);
+            const QString time = index.data(BaseLogModel::TimeRole).toString();
+            style->drawItemText(painter, textRect, option.displayAlignment, option.palette, (option.state & QStyle::State_Enabled), time);
+
+            painter->setPen(originalPen);
+            const QFontMetrics fontMetrics = painter->fontMetrics(); // option.fontMetrics adds extra padding to QFontMetrics::width
+            const int separatorLeft = horizontalAdvance(fontMetrics, time);
```

Comment:
seems the variable is badly named, I would use `separatorWidth`

Reply:
It is "separatorleft", and following one is "messageLeft"

Reply:
Hmm... I see what you mean. But "left" isn't a meaningful thing, it is actually a coordinate.

Reply:
maybe `separatorCoordinateX`?